db/silkworm/etl/collector.cpp

```diff
@@ -50,19 +49,23 @@ void Collector::collect(Entry& entry) {
     }
 }
 
-void Collector::load(silkworm::lmdb::Table* table, LoadFunc load_func) {
+void Collector::load(silkworm::lmdb::Table* table, LoadFunc load_func, unsigned int flags) {
+
     if (!file_providers_.size()) {
         buffer_.sort();
-        for (const auto& entry : buffer_.get_entries()) {
-            auto entries{load_func(entry)};
-            for (const auto& entry2 : entries) {
-                table->put(entry2.key, entry2.value);
+        for (const auto& etl_entry : buffer_.get_entries()) {
+            auto trasformed_etl_entries{load_func(etl_entry)};
+            for (const auto& transformed_etl_entry : trasformed_etl_entries) {
+                MDB_val mdb_key{db::to_mdb_val(transformed_etl_entry.key)};
+                MDB_val mdb_data{db::to_mdb_val(transformed_etl_entry.value)};
+                lmdb::err_handler(table->put(&mdb_key, &mdb_data, flags));
```

Comment:
How are these 3 lines better than `table->put(entry2.key, entry2.value);` ?

Reply:
They're not better but here I can handle flags.
The original signature for
```
void Table::put(ByteView key, ByteView data) {
    MDB_val key_val{db::to_mdb_val(key)};
    MDB_val data_val{db::to_mdb_val(data)};
    err_handler(put(&key_val, &data_val, 0));
}
```
does not implement flags which are hardcoded.
If you wish we could set the flags parameter there as defaulted to zero

Reply:
Right, let's have the flags parameter - it's probably be a net gain in code simplicity.